utils/build-script-impl

```diff
@@ -243,10 +243,6 @@ KNOWN_SETTINGS=(
     skip-merge-lipo-cross-compile-tools           ""                "set to skip running merge-lipo after installing cross-compiled host Swift tools"
     coverage-db                                   ""                "If set, coverage database to use when prioritizing testing"
     skip-local-host-install                       ""                "If we are cross-compiling multiple targets, skip an install pass locally if the hosts match"
-
-    # SWIFT_ENABLE_TENSORFLOW
-    enable-tensorflow                             ""                "If true, enable compiling with TensorFlow support"
```

Comment:
`${ENABLE_TENSORFLOW}` has a user in this file. Namely, the flag gets propagated to CMake as `SWIFT_ENABLE_TENSORFLOW`:

https://github.com/apple/swift/blob/dd38ba3215182eed1dcbd5b597815b1e2becd1cd/utils/build-script-impl#L1703

https://github.com/apple/swift/blob/dd38ba3215182eed1dcbd5b597815b1e2becd1cd/CMakeLists.txt#L80-L82

And `SWIFT_ENABLE_TENSORFLOW` in CMake determines whether `tensorflow` is added as an available feature to `lit`. AFAIK that's the only user of the CMake flag.

https://github.com/apple/swift/blob/dd38ba3215182eed1dcbd5b597815b1e2becd1cd/test/lit.site.cfg.in#L65-L68

As is, I believe lit tests that check `tensorflow` or `no_tensorflow` will fail.

---

Idea: it may be easiest to simply enable the `SWIFT_ENABLE_TENSORFLOW` CMake option by default (since it exists only on `tensorflow` branch), and remove the user of `${ENABLE_TENSORFLOW}` in `utils/build-script-impl`. This keeps `lit` tests passing.

Reply:
Cool, thanks for catching. Setting it always to true sounds good to me. I have done that.

I did a grep and there are a few other uses of the cmake build flag. I removed a few that are easily to completely remove, and there are a few left that we could remove later:
```
marcrasi@marcrasi:~/swift-base/swift$ git grep "^[^/#]*ENABLE_TENSORFLOW"
CMakeLists.txt:option(SWIFT_ENABLE_TENSORFLOW
CMakeLists.txt:if(SWIFT_ENABLE_TENSORFLOW)
CMakeLists.txt:  add_definitions(-DSWIFT_ENABLE_TENSORFLOW)
CONTRIBUTING.md:`SWIFT_ENABLE_TENSORFLOW` comment at the top of every diff hunk. This makes
README.md:<!-- SWIFT_ENABLE_TENSORFLOW -->
README.md:<!-- SWIFT_ENABLE_TENSORFLOW END -->
test/CMakeLists.txt:normalize_boolean_spelling(SWIFT_ENABLE_TENSORFLOW)
test/lit.site.cfg.in:if "@SWIFT_ENABLE_TENSORFLOW@" == "TRUE":
tools/SourceKit/cmake/modules/AddSwiftSourceKit.cmake:  if(SWIFT_ENABLE_TENSORFLOW)
utils/build-script-impl:                    -DSWIFT_ENABLE_TENSORFLOW:BOOL=TRUE
```